src/plugins/data_source/server/data_source_route_handler_context.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// eslint-disable-next-line max-classes-per-file
+import { Logger } from 'src/core/server';
+import { IDataSourceClient } from './client/data_source_client';
+
+class OpenSearchDataSourceRouteHandlerContext {
+  private logger: Logger;
+  constructor(private dataSourceClient: IDataSourceClient, logger: Logger) {
+    this.logger = logger;
+  }
+
+  public async getClient(dataSourceId: string) {
+    try {
+      const client = await this.dataSourceClient.asDataSource(dataSourceId);
+      return client;
+    } catch (error) {
+      // TODO: convert as audit log when integrate with osd auditing
+      this.logger.error(
+        `Fail to get data source client for dataSource id: [${dataSourceId}]. Detail: ${error.messages}`
+      );
+      throw error;
+    }
+  }
+}
+
+export class DataSourceRouteHandlerContext {
```

Comment:
what is the need of this wrapper? Why not export `OpenSearchDataSourceRouteHandlerContext` directly?

Reply:
Just making this flexible, maybe in the future we add other types of datas source, and this also following the `core_route_handler_context.ts` manner.